linkcheck/htmlutil/formsearch.py

```diff
@@ -37,26 +38,25 @@ def __repr__(self):
         return "<url=%s data=%s>" % (self.url, self.data)
 
 
-class FormFinder(object):
-    """Base class handling HTML start elements.
-    TagFinder instances are used as HtmlParser handlers."""
+class FormFinder(TagFinder):
+    """Class for finding links in HTML forms."""
+    TAG_FORM = "form"
+    TAG_INPUT = "input"
 
     def __init__(self):
         """Initialize local variables."""
         super(FormFinder, self).__init__()
-        # parser object will be initialized when it is used as
-        # a handler object
-        self.parser = None
         self.forms = []
         self.form = None
+        self.tag_filter = FormFinder.TAG_FORM, FormFinder.self.TAG_INPUT
```

Comment:
Minor: this would be more readable if you enclosed the tuple literal in parentheses.
```suggestion
        self.tag_filter = (FormFinder.TAG_FORM, FormFinder.self.TAG_INPUT)
```

Reply:
This is now not included in this PR and will be superseded in the next.